octopus_deploy/datadog_checks/octopus_deploy/check.py

```diff
@@ -247,10 +282,12 @@ def collect_server_nodes_metrics(self):
             self.gauge(SERVER_MAX_TASKS_METRIC, max_tasks, tags=self.base_tags + server_tags)
 
     def check(self, _):
+        self._update_completed_times()
         for _, _, project_group, _ in self.project_groups():
```

Comment:
If a new `project_group` appears, do we have a corresponding project discovery for it, as we are creating a project discovery map only during the first time the check is run?

Reply:
Good catch, pushed a fix!